### PR TITLE
server/api: omit empty field

### PR DESCRIPTION
**Pull request by nolouch**
PTAL @siddontang @disksing

### CODE REVIEW

File: pkg/typeutil/string_slice.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package typeutil
```
Comment (siddontang) [inline pkg/typeutil/string_slice.go]: please add test independently.

File: pkg/typeutil/string_slice.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+type StringSlice []string
+
+// MarshalJSON returns the size as a JSON string.
+func (s StringSlice) MarshalJSON() ([]byte, error) {
+	return []byte(`"` + strings.Join(s, ",") + `"`), nil
+}
+
+// UnmarshalJSON parses a JSON string into the bytesize.
+func (s *StringSlice) UnmarshalJSON(text []byte) error {
+	data, err := strconv.Unquote(string(text))
```
Comment (siddontang) [inline pkg/typeutil/string_slice.go]: why unquote here? you don't use quote in marshal.

Comment (nolouch) [inline pkg/typeutil/string_slice.go]: I use quote in marshal. and the string always encodes with a quote. client just distinguish number and string.

File: pkg/typeutil/string_slice_test.go [hunk 3]
```diff
@@ -14,9 +14,23 @@
...
+
+func (s *testStringSliceSuite) TestJSON(c *C) {
+	b := StringSlice([]string{"zone", "rack"})
+	o, err := json.Marshal(b)
+	c.Assert(err, IsNil)
+	c.Assert(string(o), Equals, "\"zone,rack\"")
+
+	var nb StringSlice
+	err = json.Unmarshal(o, &nb)
+	c.Assert(err, IsNil)
```
Comment (siddontang) [inline pkg/typeutil/string_slice_test.go]: check nb equals b

File: pkg/typeutil/string_slice.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+	"github.com/juju/errors"
+)
+
+//StringSlice is more friendly to json encode/decode
+type StringSlice []string
+
+// MarshalJSON returns the size as a JSON string.
+func (s StringSlice) MarshalJSON() ([]byte, error) {
+	return []byte(`"` + strings.Join(s, ",") + `"`), nil
```
Comment (disksing) [inline pkg/typeutil/string_slice.go]: Use `strconv.Quote` instead?

**Comment by disksing** [comment]
LGTM.

**Pull request merged**